codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/Decoder.java

```diff
@@ -429,23 +404,113 @@ private static IllegalArgumentException notEnoughDataException(ByteBuf in) {
     }
 
     // Unsigned Little Endian Base 128 Variable-Length Integer Encoding
-    private static int decodeULE128(ByteBuf in) throws Http2Exception {
-        final int writerIndex = in.writerIndex();
-        for (int readerIndex = in.readerIndex(), shift = 0, result = 0;
-             readerIndex < writerIndex; ++readerIndex, shift += 7) {
-            byte b = in.getByte(readerIndex);
-            if (shift == 28 && (b & 0xF8) != 0) {
-                in.readerIndex(readerIndex + 1);
-                break;
-            }
-
-            if ((b & 0x80) == 0) {
+    private static int decodeULE128(ByteBuf in, int result) throws Http2Exception {
```

Comment:
Random thoughts: 

- The C1 compiler uses bytecode size of a function to decide if it should be inlined.   If it was small and not called very often I think it might have had a difference.  Meh.
- Java8 more aggressively evicts compiled code from its cache which might hurt when it starts looking for large functions.  Again, meh.